### PR TITLE
#3777 Fix lint errors in notificationSubscriptionHander.js

### PR DESCRIPTION
**Pull request by blascsi**
## What type of PR is this? (check all applicable)

- [X] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Fix linting issues found in notificationSubscriptionHandler.js

## Related Tickets & Documents
#3777 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [X] no documentation needed

### CODE REVIEW

**Comment by blascsi** [comment]
Thanks! Happy to help!

File: app/javascript/packs/notificationSubscriptionHandler.js [hunk 1]
```diff
@@ -42,23 +42,22 @@ function loadFunctionality() {
...
         allButtons[i].classList.remove('selected');
       }
       if (shouldUnsubscribeToNotifications) {
-        const unsubscribeButton = allButtons.namedItem(
-          'unsubscribe',
-        );
+        const unsubscribeButton = allButtons.namedItem('unsubscribe');
         unsubscribeButton.classList.add('selected');
-        payload = unsubscribeButton.dataset.payload;
+        ({ dataset: payload } = unsubscribeButton);
```
Comment (maestromac) [inline app/javascript/packs/notificationSubscriptionHandler.js]: The subscription feature is broken with this change. I believe you meant to do 
```suggestion
        ({ payload } = unsubscribeButton.dataset)
```
With that, I was able to get subscription working. Is this a new ECMAScript feature?

File: app/javascript/packs/notificationSubscriptionHandler.js [hunk 2]
```diff
@@ -42,23 +42,22 @@ function loadFunctionality() {
...
       } else {
         target.classList.add('selected');
-        payload = target.dataset.payload;
+        ({ dataset: payload } = target);
```
Comment (maestromac) [inline app/javascript/packs/notificationSubscriptionHandler.js]: Similar to above
```suggestion
        ({ payload } = target.dataset)
```

File: app/javascript/packs/notificationSubscriptionHandler.js (hunk 1 above)
Comment (jacobherrington) [reply app/javascript/packs/notificationSubscriptionHandler.js]: Good catch, I missed this one! 😅

Comment (blascsi) [reply app/javascript/packs/notificationSubscriptionHandler.js]: Sorry about that. This was the first thing that I tried, but ESLint was giving me a hard time about it still, but now it seems to work.
Or I might have just looked at the wrong output from the linter. 😅
Anyway I made the commit with the recommended fixes.

File: app/javascript/packs/notificationSubscriptionHandler.js (hunk 2 above)
Comment (blascsi) [reply app/javascript/packs/notificationSubscriptionHandler.js]: Committed the recommended fixes here too.

File: app/javascript/packs/notificationSubscriptionHandler.js (hunk 1 above)
Comment (jacobherrington) [reply app/javascript/packs/notificationSubscriptionHandler.js]: No worries! I'll take a look at this PR after I grab some 🍣

**Comment by rhymes** [comment]
@jacobherrington @maestromac please keep this updated once it's merged https://github.com/thepracticaldev/dev.to/issues/3777

great job everyone! 🥳

**Comment by blascsi** [comment]
Thank you everyone! You are awesome!

**Pull request merged**